install.sh

```diff
@@ -9,43 +10,130 @@ APP_BINARY="/usr/bin/xochitl"
 
+uninstall () {
     rm -rf ${CACHE_DIR}*
+}
 
-    echo "The patch should be uninstalled now..."
-    echo -e "${COLOR_YELLOW}Please restart your device to see effects...${NOCOLOR}"
-    echo "Have fun, until next time, bye!"
 
-    exit 0
-fi
+find_version () {
+    case $hash in
+        "143aa1d2f25affbd9ee437bc1418d6f1d577b125")
+            patch_version="0.0.1"
+            ;;
+        "610c8f928ee8908faa9cd7439271c46985952a30")
+            patch_version="0.0.1"
+            ;;
+        "ae0d21c258c0f3d93717d9bd23eb74b68ac438db")
+            patch_version="0.0.3"
+            ;;
+        "4d066636ed653ffe59d4bc3acf55aa6cef72d795")
+            patch_version="0.0.3"
+            ;;
+    esac
+}
+
+
+patch () {
+    hash=$(sha1sum $APP_BINARY | cut -c1-40)
+
+    if [ -z "$patch_version" ]; then
+        find_version
+    fi
 
-hash=$(sha1sum $APP_BINARY | cut -c1-40)
-case $hash in
-    "143aa1d2f25affbd9ee437bc1418d6f1d577b125")
-        patch_version="0.0.1"
-        ;;
-    "610c8f928ee8908faa9cd7439271c46985952a30")
-        patch_version="0.0.1"
-        ;;
-    "ae0d21c258c0f3d93717d9bd23eb74b68ac438db")
-        patch_version="0.0.3"
-        ;;
-    "4d066636ed653ffe59d4bc3acf55aa6cef72d795")
-        patch_version="0.0.3"
-        ;;
-    *)
-        echo -e "${COLOR_RED}No suitable patch found for your version :(${NOCOLOR}"
+    if [ -z "$patch_version" ]; then
+        echo -e "${COLOR_RED}No suitable patch found...${NOCOLOR}"
         exit 1
-        ;;
-esac
+    fi
+
+
+    echo -e "${COLOR_GREEN}Trying to download and install patch: '$patch_version'${NOCOLOR}"
+
+    pass=$(sha256sum $APP_BINARY | cut -c1-64)
+    wget -O- $PATCH_URL/${patch_version}_$hash.patch | openssl aes-256-cbc -d -a -md sha512 -pbkdf2 -iter 1000000 -salt -pass pass:$pass | tar --overwrite -xjC $CACHE_DIR
+}
+
+
+ask () {
+    while true; do
+        echo -ne "${COLOR_BLUE}"
+        read -p "$1 [Y/n]? " yn
+        echo -ne "${NOCOLOR}"
+        case $yn in
+            [Yy]* ) return 0;;
+            [Nn]* ) return 1;;
+            "" ) return 0;;
+        esac
+    done
+}
+
+
+install_stylus () {
+    set +e
+    ask "Would you like to install 'ddvk/remarkable-stylus'"
+    resp=$?
+    set -e
+    if [ "$resp" -eq "0" ]; then
+        sh -c "$(wget https://raw.githubusercontent.com/ddvk/remarkable-stylus/master/patch.sh -O-)"
+    fi
+}
+
+
+set_timezone () {
+    echo -ne "${COLOR_BLUE}Your current timezone is: "
+    timedatectl | grep "Time zone" | xargs | cut -c12-
+    echo -ne "${NOCOLOR}"
+    set +e
+    ask "Would you like to set proper timezone"
+    resp=$?
+    set -e
+    if [ "$resp" -eq "0" ]; then
+
+        echo "Possible timezones are:"
+        echo "-----------------------------------------------------------------------------------------------------------------------------------------------------------"
+        timedatectl list-timezones | xargs -n5 | awk '{printf "%-30s %-30s %-30s %-30s %-30s\n",$1,$2,$3,$4,$5 }'
```

Comment:
Just to warn you, this list will return more than is actually available on the device.
https://remarkable.guide/guide/config/timezone.html#available-timezones

Reply:
Are you sure about that? There is no information saying directly that `timedatectl list-timezones` will not return only available ones...

Reply:
> Are you sure about that? There is no information saying directly that `timedatectl list-timezones` will not return only available ones...

I guess I should update the page to say that. Yes I'm sure, that's why the page tells you to inspect the filesystem, and then install the full list before you can use `timedatectl list-timezones`

Reply:
I've updated the page to clarify it.

Reply:
Ok, good to know, I should then improve it by checking if there is proper dir/file in `/usr/share/zoneinfo`.
Thanks!

Reply:
It might also be worth trying to detect if toltecctl is installed and starting to try to make your script play nice with a toltec install. I know toltec doesn't support past 2.15.1 yet, but when people finally get around to starting to approve and merge my PRs, the plan is to get it updated.